transport/gdrive-transport/src/main/java/org/apache/airavata/mft/transport/gdrive/GDriveSender.java

```diff
@@ -0,0 +1,121 @@
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+
+
+public class GDriveSender implements Connector {
+    private static final Logger logger = LoggerFactory.getLogger(GDriveSender.class);
+
+    private GDriveResource gdriveResource;
+    private Drive drive;
+    private JsonObject jsonObject;
+
+
+    @Override
+    public void init(String resourceId, String credentialToken, String resourceServiceHost, int resourceServicePort, String secretServiceHost, int secretServicePort) throws Exception {
+        ResourceServiceGrpc.ResourceServiceBlockingStub resourceClient = ResourceServiceClient.buildClient(resourceServiceHost, resourceServicePort);
+        this.gdriveResource = resourceClient.getGDriveResource(GDriveResourceGetRequest.newBuilder().setResourceId(resourceId).build());
+
+        SecretServiceGrpc.SecretServiceBlockingStub secretClient = SecretServiceClient.buildClient(secretServiceHost, secretServicePort);
+        GDriveSecret gdriveSecret = secretClient.getGDriveSecret(GDriveSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
+
+        HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = new JacksonFactory();
+        String jsonString = gdriveSecret.getCredentialsJson();
+        jsonObject = new JsonParser().parse(jsonString).getAsJsonObject();
+        GoogleCredential credential = GoogleCredential.fromStream(new ByteArrayInputStream(jsonString.getBytes(StandardCharsets.UTF_8)), transport, jsonFactory);
+
+        if (credential.createScopedRequired()) {
+            Collection<String> scopes = DriveScopes.all();
+            credential = credential.createScoped(scopes);
+        }
+
+        drive = new Drive.Builder(transport, jsonFactory, credential).build();
+
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public void startStream(ConnectorContext context) throws Exception {
+        logger.info("Starting GDrive send for remote server for transfer {} ", context.getTransferId());
+        String id = null;
+        boolean fileupdated = false;
+        String entityUser = jsonObject.get("client_email").getAsString();
+        File fileMetadata = new File();
+
+        InputStreamContent contentStream = new InputStreamContent(
+                "", context.getStreamBuffer().getInputStream());
+
+
+        fileMetadata.setName(this.gdriveResource.getResourcePath());
+        FileList fileList = drive.files().list()
+                .setQ("name = '"+gdriveResource.getResourcePath()+"'")
+                .setFields("files(id,name)")
+                .execute();
+
+        for (File f : fileList.getFiles()) {
+                id = f.getId();
+                File file = drive.files().get(id).execute();
+                drive.files().update(file.getId(), fileMetadata, contentStream).setFields("id").execute();
+                fileupdated = true;
+        }
+
+        if (fileupdated == false) {
```

Comment:
What is this logic doing?

Reply:
So the logic above, checks if the file exists with that name.
If it does it updates the file else it creates a new file.

Reply:
I think it's better to update always by fileId. For both box and Google drive, we can expect the files(both source and destination) to be available already for the transactions. Otherwise, it'll be confusing. This is just my view.

Reply:
Yes updation is based on fileID only, but filter is on the basis of fileName as that would be the input from user.

Reply:
Remove this section and use the method I suggested in above comment

Reply:
Changes added and tested!